data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/tracer/StateMachineTracer.java

```diff
@@ -25,4 +25,7 @@
  * The tracer allows to observe triggered event (before) and transitions (after) of the {@link OpensearchStateMachine}
  */
 public interface StateMachineTracer extends Trace<OpensearchState, OpensearchEvent> {
+
+    void setStateMachine(OpensearchStateMachine stateMachine);
```

Comment:
Making this method default (with empty implementation) would remove the need to implement it in most of the tracers where we don't need it anyway.